lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1229,7 +1275,37 @@ func (d *zfs) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, al
 
 // DeleteVolume deletes a volume of the storage device. If any snapshots of the volume remain then
 // this function will return an error.
+// For image volumes, both filesystem and block volumes will be removed.
 func (d *zfs) DeleteVolume(vol Volume, op *operations.Operation) error {
+	for _, suffix := range blockBackedAllowedFilesystems {
```

Comment:
We only need to do this loop for image volumes that are `isBlockBacking` right?

Reply:
Yes. We could guard it with `if vol.volType == VolumeTypeImage && d.isBlockBacking(vol)`, if that's what you're thinking ;)

Reply:
Yes I think that would be clearer, and more efficient, as otherwise we are going to be unnecessarily checking if volumes exist that cannot possible exist.